crates/typst/src/math/accent.rs

```diff
@@ -114,7 +114,12 @@ impl LayoutMath for Packed<AccentElem> {
     #[typst_macros::time(name = "math.accent", span = self.span())]
     fn layout_math(&self, ctx: &mut MathContext, styles: StyleChain) -> SourceResult<()> {
         let cramped = style_cramped();
-        let base = ctx.layout_into_fragment(self.base(), styles.chain(&cramped))?;
+        let mut base = ctx.layout_into_fragment(self.base(), styles.chain(&cramped))?;
+
+        // Try replace a glyph with its dotless variant.
+        if let MathFragment::Glyph(ref mut glyph) = &mut base {
+            glyph.make_dotless_form(ctx)
```

Comment:
`dtls` feature should not be applied unconditionally, for example people might want $\hat{\imath}$ but $\widehat{\dot{\imath}+\dot{\jmath}}$. LaTeX has `\imath` and `\jmath` macros to explicitly request the dotless form.

Reply:
Are there any situations where you'd want an accent above i or j, and keep the dot? Also, I think Typst has `dotless.i` and `dotless.j` in `sym` that correspond to the LaTeX `\imath` and `\jmath` macros. We should add a function though so that we get the dotless variants of the blackboard bold, caligraphic, etc. variants. I'm not sure how LaTeX does this.

Reply:
I feel like I would expect those to just be available as `$bb(dotless.i)$`?

Reply:
The problem is that `dotless.i` is a Unicode char, but the variants in the `dtls` feature are from the font. I don't think there exist codepoints for blackboard bold, caligraphic etc. dotless i and j.

Reply:
Well could we look those up when we convert to a glyph in `styled_char` and give a warning if they don't exist? I wouldn't want a user to have to care about other functions if it lets us provide a consistent interface.

Reply:
LaTeX does not use `dtls` feature yet (https://github.com/latex3/unicode-math/issues/34). Since fonts might be inconsistent here, `dotless.i` and `dotless.j` should check if the font has `dtls` feature and make them the equivalent of `i` and `j` with `dtls` feature applied (i.e. they would switch alphabets when requested), otherwise use the dotless i and j Unicode code points.